src/app/shared/pipes/truncate.pipe.ts

```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'truncate',
+})
+export class TruncatePipe implements PipeTransform {
+  transform(value: string, limit = 20, trail = '...'): string {
+    const limited = value.substring(0, limit);
+    return value.length > limit ? limited.slice(0, limited.lastIndexOf(' ')) + trail : value;
```

Comment:
```suggestion
      const length = value.length;
      return length > limit ? limited.slice(0, length) + trail : value;
```
but I could also not see the reson of using lastIndexOf

Reply:
Powodem jest to żeby wyraz nie ucinał się w połowie tylko na ostatniej spacji przed 200 znakami

Reply:
ok. No to jasne. 
```suggestion
    return value.length > limit ? limited.slice(0, limited.lastIndexOf(' ')) + trail : value;
    
    if (limited.length < 0 limit) {
      return value;
    }
    
    return limit ? limited.slice(0, limited.lastIndexOf(' ')) + trail
```
To tak, zeby uniknąć niepotrzebnego obliczania `const limited` w przypadku kiedy mieścimy się w tym limicie.

Reply:
<img width="700" alt="image" src="https://user-images.githubusercontent.com/12514322/161726003-90ae5f95-3488-4dc4-b374-f9aede8ead3b.png">

To ja bym proponował tak:
`export class TruncatePipe implements PipeTransform {
  transform(value: string, limit = 20, trail = '...'): string {
    const limited = value.substring(0, limit);
    if (value.length > limit) {
      return limited.slice(0, limited.lastIndexOf(' ')) + trail;
    } else {
      return value;
    }
  }
}`